### DISCUSSION TITLE
A DX7 USB Dongle

### DISCUSSION THREAD

**Discussion by diyelectromusic** [probonopd/MiniDexed#568] [Show and tell]
Using the new [USB Gadget Mode](https://github.com/probonopd/MiniDexed/wiki/Hardware#usb-gadget-mode) support for MiniDexed, I've written up how to use it with a Pi Zero to build a DX7 USB Dongle.  Just plug it into your PC and an audio output and you've a USB MIDI DX7 (or 8 if you're using a Zero 2W) ready to go :)

And no, I don't know why you'd not just install Dexed either really, but I still think there is something a bit neat about having a $2000 80s synth on a $20 USB stick...

https://diyelectromusic.wordpress.com/2023/11/22/a-dx7-usb-dongle/

Kevin

![IMG_7469](https://github.com/probonopd/MiniDexed/assets/68612569/c6743b94-ed5c-431c-a1a3-241d40b5ece7)

**Comment by diyelectromusic** [comment]
By the way, this hit Hackaday... https://hackaday.com/2023/11/24/a-yamaha-dx7-on-a-usb-dongle/

Kevin

**Reply by diyelectromusic** [reply]
Although some of the comments make you wonder what is going on sometimes...

> The line out port is facing the worst possible direction

Yep, that is why I'm actually using the one as shown above (from my blog post) with the Zero STEM because funnily enough I did notice that too...

> And boy, look at the usb-c disaster to feed the thing.

Ah yes.. er what?  I guess if you try to power via USB-C then it would be a bit of a disaster seeing as there is no USB-C on a Zero...

> But on the positive side, a dx7 in compact form is a nice thing to have, be it monophonic.

8 or 16 note polyphony actually, but anyway...

> In the main picture, it looks like getting audio out and powering it at the same time might be a challenge.

Hey you know what?  I just stuck it on top, took a photo and never tried to plug it in - but you're right.  So glad you pointed that out.

JK - actually, no, funnily enough it works fine, even with the no-solder USB STEM.

> It is a Pi, it runs Linux, it can emulate things.

Ah yes, a bit like the universal Turing machine - once Linux is running, all possible projects have now been solved and aren't worth mentioning anymore.  Did I mention it wasn't Linux?  No?  Ah.  Well it isn't.

> This has been done on lower forms of board life like arduino blue pill with Mozzi but still glad to see it has somehow inspired others.

True, but irrelevant.  (Aside, blue-pill is an STM32 based board and not Arduino, but let's not worry about factual details).  Mozzi is excellent of course, but it is just a code library for synthesis, including FM.  As much in common with a DX7 as suggesting a train and a car are the same as they both have engines and wheels... anyway, moving on...

Yes, it's amazing how inspiring that Mozzi one-carrier, one-modulator, non-MIDI, no-envelope, monophonic, PWM, no-user-interface, FMSynth.ino example can be...

> The issues I see is that 1) USB has been phased out in favor of USB-C cabling/ports and

Damn, there was me with it all wired up and no USB A ports to plug it into... oh well, back to the drawing board...

> 2) latency due to lower USB-A’s transfer rates.

USB really struggles to keep up with those 31,250 serial baud rates for MIDI.  What do you mean first generation USB MIDI is even faster?  Wow, when did that come out then?  Oh, November 1999.  So did the announcement really say "USB can easily handle heavy loads of MIDI data while preserving the timing integrity of the data. Hundreds of MIDI note messages can be sent all at the same time. In addition, by handling multiple "virtual" cables the USB offers a solution to go beyond MIDI's 16-channel limit." Hang on, USB 2.0 didn't come out until year 2000, so was this referring to USB 1.0 full speed 12MB/s links?

Yes, turns out it was :) https://www.midi.org/specifications-old/item/usb-midi-1-0-specification

Apparently 12MB/s is a lot faster than 31.25 KB/s - who'd have thought it...

Kevin

**Comment by diyelectromusic** [comment]
I now have a Pi Zero shaped IO module too that is particularly suited to being used with the dongle.  Details here: https://diyelectromusic.wordpress.com/2023/12/02/rpi-zero-minidexed-io-board-pcb-design/

Kevin

![IMG_7492](https://github.com/probonopd/MiniDexed/assets/68612569/ee849e42-d9f5-465c-a2a3-b97e404d9bba)

**Comment by BobanSpasic** [comment]
Hi Kevin,
I am trying to get a Zero 2W working as a USB dongle, but no luck. I am using the Addon board v1.1 from Amazon (I also saw it on your blog) and a generic PCM5102.
If I connect the dongle without SD-Card to my PC, I get a BCM2710 Boot device in Device Manager. The HDMI display shows nothing.
If I connect the dongle with a SD-Card (with Minidexed on it), I get nothing in Device Manager, and HDMI display also shows nothing.
Any idea how to debug anything here?

**Reply by diyelectromusic** [reply]
Sounds like it isn't finding a proper kernel on the SD card maybe...?  If you power the Zero independently via the PWR USB socket, i.e. not connected to the PC, do you see anything on HDMI?

**Reply by BobanSpasic** [reply]
Nothing changes. I am installing LibreELEC to SD-Card at the moment. Let see if that is working.

**Reply by BobanSpasic** [reply]
Well, not working, but at least an error on the HDMI screen with LibreELEC:
*** Error in mount_sysroot: mount_common: Could not mount /flash/SYSTEM ***

Probably the LibreELEC-Version is too old and does not support this hardware

**Reply by BobanSpasic** [reply]
Raspbian (Debian Bullseye) 32bit is working. I am at least at ease that my hardware is OK.

**Reply by diyelectromusic** [reply]
Sorry, at this point I'm not sure what else to suggest.  I've just downloaded and booted the last continuous release on a Zero 2W and all seems fine...?

![IMG_7688](https://github.com/probonopd/MiniDexed/assets/68612569/36730f34-be74-4d9f-93cd-b62ec5ec7b82)

**Reply by BobanSpasic** [reply]
Could you upload somewhere your build for me?
I have tested with the latest official build. I've edited minidexed.ini to match my config.

[edit] I see now - you wrote you've used the latest official build.

**Reply by BobanSpasic** [reply]
I apologize about a late update - a new SD Card did the trick. The old one isn't malfunctioning, it is just not recognized by MiniDexed/Circle. Linux is booting just fine from the same SD Card
